File: test/data/cs/test04/StateInitial.cs
public class StateInitial : BaseState
{
    private StateController _stateController; 
    private IControllee _controllee; 
    public StateInitial(StateController stateController, IControllee controllee) : base(controllee)
    {
        _stateController = stateController;
        _controllee = controllee;
    }
    public override BaseState? TransitGoInTo()
    {
        _controllee.DoAction0();
        return _stateController.InstanceOfIdle; 
    }
    public override string GetStateName()
    {
        return "Initial"; 
    }
}

File: test/data/cpp/test04/StateInitial.cpp
#include "StateInitial.h"
StateInitial::StateInitial(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
StateInitial::~StateInitial()
{
}
BaseState* StateInitial::TransitGoInTo()
{
    _controllee->DoAction0();
    _stateController->InstanceOfNotShooting->Setup(false, false);
    return _stateController->InstanceOfNotShooting;
}
const char* StateInitial::GetStateName()
{
    return "Initial";
}
BaseState* StateInitial::GetParent()
{
    return 0;
}
int StateInitial::GetStateID()
{
    return 3;
}